### DISCUSSION TITLE
Error assigning role to service principal, exiting.

### DISCUSSION THREAD

**Discussion by yeggan** [Azure-Samples/graphrag-accelerator#94] [Q&A] [answered]
Hello everyone,

I am the Owner of a Subscription and an RBAC Administrator of the Subscription, but I do not have Admin rights in Microsoft Entra. I followed the steps in the [DEPLOYMENT-GUIDE](https://github.com/Azure-Samples/graphrag-accelerator/blob/main/docs/DEPLOYMENT-GUIDE.md) for the graphrag-accelerator project.

Here are the steps I took:

1. Opened the repository through Codespaces.
2. Created the resource group.
3. Filled out infra/deploy.parameters.json.
4. Deployed the solution accelerator to the resource group.

After approximately 40 minutes, I received the following error:
``Deploying Azure resources...
Deployment name: graphrag-deploy-2024-07-18_05-51-57
Assigning 'Cognitive Services OpenAI Contributor' AOAI role to managed identity...
Error assigning role to service principal, exiting...``

When I checked the Azure Portal, I noticed that 35 resources were created in the initial resource group (rg-graphrag-accelerator), and 9 resources were created in a new resource group (MC_rg-graphrag-accelerator).

I would appreciate any guidance on how to resolve this issue. Is there a specific permission or role I might be missing due to my lack of Admin rights in Microsoft Entra? Any help or pointers would be greatly appreciated!

**Comment by timothymeyers** [comment, marked as answer]
Hi @yeggan - is your Azure Open AI Service deployed in the same subscription that you are doing the accelerator deployment in?

**Comment by SridharArrabelly** [comment]
@timothymeyers  @yeggan is there a resolution for this? i am having the same exact problem. my openai resource is in the same subscription as accelerator deployment.
